src/tests/testURL.cc

```diff
@@ -61,3 +61,64 @@ testURL::testDefaultConstructor()
     delete urlPointer;
 }
 
+void
+testURL::benchmarkEncoder()
```

Comment:
Will define it out later. I applied it for now so we can get a view of what the different farm OS have for performance results. I am thinking as a followup adding a "make benchmark" build option to build and run this and others we want to add in future.

Reply:
> Will define it out later

If you really want to temporary keep this code in the PR, please feel free to keep it. We just need to remember to remove it (and not just "define it out") before the merge.

> I am thinking as a followup adding a "make benchmark" build option to build and run this and others we want to add in future.

It is very difficult to write meaningful benchmarking code/harness, the current unit test infrastructure is a poor substrate for doing that, and it is likely that using pull request review to reach a good outcome will be very painful. If you really want to work in the direction of adding micro-level benchmarks, let's start with an architecture-level proposal rather than code changes.

Reply:
I'm aware. Which is why this is not trying to add a test harness or anything like that. Just a simple micro-bench for relative timing comparison of proposed changes. As of 7fc9da8 the build option USE_BENCHMARK needs to be #define'd to build with the test.

Reply:
I am glad that we agree on the lack of appropriate test harness for micro performance tests. That agreement makes my original change request even stronger. I can also critique the added code itself, but I hope that will not be necessary because it should not be in the official repository for the reasons already stated.